packages/library/src/airdrop-confirmer.ts

```diff
@@ -0,0 +1,60 @@
+import { GetSignatureStatusesApi } from '@solana/rpc-core/dist/types/rpc-methods/getSignatureStatuses';
+import { SignatureNotificationsApi } from '@solana/rpc-core/dist/types/rpc-subscriptions/signature-notifications';
+import { Rpc, RpcSubscriptions } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import { TransactionSignature } from '@solana/transactions';
+
+import { BaseTransactionConfirmationStrategyConfig, raceStrategies } from './transaction-confirmation-strategy-racer';
+import { createRecentSignatureConfirmationPromiseFactory } from './transaction-confirmation-strategy-recent-signature';
+import { getTimeoutPromise } from './transaction-confirmation-strategy-timeout';
+
+interface DefaultSignatureOnlyRecentTransactionConfirmerConfig {
+    rpc: Rpc<GetSignatureStatusesApi>;
+    rpcSubscriptions: RpcSubscriptions<SignatureNotificationsApi>;
+}
+
+interface WaitForRecentTransactionWithTimeBasedLifetimeConfirmationConfig
+    extends BaseTransactionConfirmationStrategyConfig {
+    getTimeoutPromise: typeof getTimeoutPromise;
+    signature: TransactionSignature;
+}
+
+/** @deprecated */
```

Comment:
Any reason these are deprecated?

Reply:
Yeah! Totally. Check out this conversation on Discord: https://discord.com/channels/428295358100013066/439194979856809985/1164344045842214953

The blockheight-based transaction confirmation strategy was created ~1 year ago in reaction to the timeout-based strategy missing the point. The timeout is _always_ too long to wait, because the real expiry happens when the blockhash becomes too old to land the transaction with.

Even though I didn't export these methods from the library, I wanted to mark them deprecated and then later come back and document them as such to make it extra clear not to use them if possible.